### PR TITLE
Update Rails application config defaults to 7.2

### PR DESCRIPTION
**Pull request by Nitemaeric**
## Context

We've updated our application to Rails 7.2, but we haven't updated the configurations. We want to update our application configurations to 7.2.

## Changes proposed in this pull request

- Updates Rails configuration defaults to 7.2
- Splits out smoke tests from the rest of the specs. A `smoke_test_helper.rb` file has been added to facilitate this setup.

## Guidance to review

> Set `config.action_controller.raise_on_missing_callback_actions = false`. This is because we currently set `after_action :verify_policy_scoped, only: %i[index]` in the `Placements::ApplicationController`. If left as the default, `true`, many placements system specs will fail.

This is _maybe_ something that needs addressing in the future.

### CODE REVIEW

**Review by ollietreend — COMMENTED** [review]
Nice work @Nitemaeric 🙌🏻 thanks for making these config changes!

File: config/application.rb [hunk 1]
```diff
@@ -21,7 +21,12 @@
 module IttMentorServices
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
-    config.load_defaults 7.0
+    config.load_defaults 7.2
+
+    # Please, add to the `ignore` list any other `lib` subdirectories that do
+    # not contain `.rb` files, or that should not be reloaded or eager loaded.
+    # Common ones are `templates`, `generators`, or `middleware`, for example.
+    config.autoload_lib(ignore: %w[assets tasks])
```
[the remark this answers is not in the document]
Comment (Nitemaeric) [reply config/application.rb:29]: https://github.com/DFE-Digital/itt-mentor-services/pull/1032 Let's remove the file entirely :)

File: bin/smoke [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+bin/rspec spec/smoke_tests --tag smoke_test -b
```
[the remark this answers is not in the document]
Comment (Nitemaeric) [reply bin/smoke]: Oh, I see you've recommended 1). I'll do _that_ instead 😅

[the remark this answers is not in the document]
Comment (Nitemaeric) [reply bin/smoke]: I've done a combination actually. I've added the `rspec` binstub in and also removed the `bin/smoke` file.

**Review by ollietreend — APPROVED** [review]
Looks great. Thanks!

**Pull request merged**